src/linpred.jl

```diff
@@ -101,11 +101,11 @@ mutable struct DensePredChol{T<:BlasReal,C} <: DensePred
     scratchm1::Matrix{T}
     scratchm2::Matrix{T}
 end
-function DensePredChol(X::StridedMatrix, pivot::Bool)
+function DensePredChol(X::AbstractMatrix, pivot::Bool)
     F = Hermitian(float(X'X))
     T = eltype(F)
     F = pivot ? pivoted_cholesky!(F, tol = -one(T), check = false) : cholesky!(F)
-    DensePredChol(AbstractMatrix{T}(X),
+    DensePredChol(Matrix{T}(X),
```

Comment:
Yeah, but I was wondering whether `X` may be the matrix passed by the user, or whether it's always an object that GLM creates.

Reply:
for the `lm(X::AbstractMatrix, y::AbstractVector)` method, it's passed directly by the user, right? It's only something GLM creates for the method with a formula.

Reply:
It might be interesting to add a kwarg `copy=true` to this, that we can then change to `copy=false` when calling this from the methods with a formula.